### PR TITLE
Bundle the API Blueprint parser

### PR DESCRIPTION
**Pull request by honzajavorek**
Should prevent `protagonist` to ever get to the distributed package. Previous experiments with lockfiles didn't work. Also adding a complex smoke test to verify all the scenarios in the npm package life cycle.

### CODE REVIEW

**Comment by honzajavorek** [comment]
Interesting, it passed locally. Will investigate.

**Comment by honzajavorek** [comment]
Fixed 🎉 

So now a local `npm install` to the Dredd Transactions codebase stays with the compilation (can only be avoided using `--no-optional` everywhere during development), but any manipulation with the distributed package should be completely without compilation, which is an improvement to the current state. That also means developing Dredd's codebase should be without compilation.

This also "hardcodes" only a few dependencies (a single Fury adapter and its friends) instead of all the Dredd Transactions' dependencies, so the dependents should have lighter lockfiles, I guess.

@kylef @realityking Let me know what you think about this. Kinda hack, but seems to work.

File: scripts/smoke.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      "the 'protagonist' library (a C++ binding for the API Blueprint"\
+      "parser), which is an unwanted behavior of the installation process."\
+      "The packaging of Dredd Transactions should have prevented this."
+    echo "======================================================================"
+    exit 1
+  else
+    echo "SUCCESS"
+    echo "======================================================================"
+  fi
+fi
```
Comment (kylef) [inline scripts/smoke.sh]: Should we perhaps make this script fail if we get past this if condition? If TRAVIS env was unset this script could silently fail and potentially lead to this test not running if we moved CI provider or if for whatever reason the env variable wasn't set on TRAVIS.

```suggestion
else
  exit 1
fi
```

Comment (honzajavorek) [reply scripts/smoke.sh]: Yes. This would be relevant for Dredd's smoke test as well.

**Pull request merged**